app/main.py

```diff
@@ -2,24 +2,85 @@
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: any, name: any) -> None:
+        self.protected_name = "_" + name
+
+    def __get__(self, obj: any, objtype: any = None) -> None:
+        return getattr(obj, self.protected_name)
+
+    def __set__(self, obj: any, value: int) -> None:
+        self.validate(value)
+        setattr(obj, self.protected_name, value)
+
+    def validate(self, value: int) -> None:
+        if not isinstance(value, int):
+            raise TypeError("Expected value to be an int")
+
+        if value < self.min_amount or value > self.max_amount:
+            raise ValueError("Value error !!!")
```

Comment:
Make the error message more descriptive rather than duplicating the actual error name